ts/features/payments/checkout/store/selectors/paymentMethods.ts

```diff
@@ -19,13 +21,17 @@ export const walletPaymentEnabledUserWalletsSelector = createSelector(
   walletPaymentUserWalletsSelector,
   userWalletsPot =>
     pot.map(userWalletsPot, userWallets =>
-      userWallets.filter(wallet =>
-        wallet.applications.find(
-          app =>
-            app.name === WalletApplicationNameEnum.PAGOPA &&
-            app.status === WalletApplicationStatusEnum.ENABLED
-        )
-      )
+      userWallets.filter(wallet => {
+        const walletDetails = wallet.details as UIWalletInfoDetails;
```

Comment:
Because the type varies based on the "type" field, and to decouple from that field and create ad-hoc logic for each payment method, we have created this additional type `UIWalletInfoDetails` which is the union of the various attributes regardless of the "type" field that is returned to us, making the app agnostic.